homeassistant/components/camera/__init__.py

```diff
@@ -138,23 +141,72 @@ async def async_request_stream(hass: HomeAssistant, entity_id: str, fmt: str) ->
     return await _async_stream_endpoint_url(hass, camera, fmt)
 
 
-@bind_hass
-async def async_get_image(
-    hass: HomeAssistant, entity_id: str, timeout: int = 10
+async def _async_get_image(
+    camera: Camera,
+    timeout: int = 10,
+    width: int | None = None,
+    height: int | None = None,
 ) -> Image:
-    """Fetch an image from a camera entity."""
-    camera = _get_camera_from_entity_id(hass, entity_id)
+    """Fetch a snapshot image from a camera.
 
+    If width and height are passed, an attempt to scale
+    the image will be made on a best effort basis.
+    Not all cameras can scale images or return jpegs
+    that we can scale, however the majority of cases
+    are handled.
+    """
     with suppress(asyncio.CancelledError, asyncio.TimeoutError):
         async with async_timeout.timeout(timeout):
-            image = await camera.async_camera_image()
-
-            if image:
-                return Image(camera.content_type, image)
+            # Calling inspect will be removed in 2022.1 after all
+            # custom components have had a chance to change their signature
+            sig = inspect.signature(camera.async_camera_image)
+            if "height" in sig.parameters and "width" in sig.parameters:
+                image_bytes = await camera.async_camera_image(
+                    width=width, height=height
+                )
+            else:
+                _LOGGER.warning(
+                    "The camera entity %s does not support requesting width and height, please open an issue with the integration author",
+                    camera.entity_id,
+                )
+                image_bytes = await camera.async_camera_image()
+
+            if image_bytes:
+                content_type = camera.content_type
+                image = Image(content_type, image_bytes)
+                if (
+                    width is not None
+                    and height is not None
+                    and "jpeg" in content_type
+                    or "jpg" in content_type
```

Comment:
I think we need a parenthesis around the jpeg or jpg checks.

https://docs.python.org/3/reference/expressions.html#operator-precedence

Reply:
Good catch, that would have been very difficult to track down later.

Reply:
https://github.com/home-assistant/core/pull/54474